lib/project.ts

```diff
@@ -707,100 +710,26 @@ export class Project implements Project.IProject {
+			this.saveProject(this.projectDir).wait();
 	}
-
-	private readProjectData(): IFuture<void> {
-		return (() => {
-			let projectDir = this.getProjectDir().wait();
-			let shouldSaveProject = false;
-			if (projectDir) {
-				let projectFilePath = path.join(projectDir, this.$staticConfig.PROJECT_FILE_NAME);
-				try {
-					let data = this.$fs.readJson(projectFilePath).wait();
-					if (data.projectVersion && data.projectVersion.toString() !== "1") {
-						this.$errors.fail("FUTURE_PROJECT_VER");
 					}
 
-					if (!_.has(data, "Framework")) {
-						if (_.has(data, "projectType")) {
-							data["Framework"] = data["projectType"];
-							delete data["projectType"];
-						} else {
-							data["Framework"] = TARGET_FRAMEWORK_IDENTIFIERS.Cordova;
+	private getProjectRelativePath(fullPath: string, projectDir: string): string {
+		projectDir = path.join(projectDir, path.sep);
+		if (!_.startsWith(fullPath, projectDir)) {
+			throw new Error("File is not part of the project.");
 						}
 
-						shouldSaveProject = true;
-					}
-
-					this.projectData = data;
-					this.frameworkProject = this.$frameworkProjectResolver.resolve(this.projectData.Framework);
-					shouldSaveProject = this.$projectPropertiesService.completeProjectProperties(this.projectData, this.frameworkProject) || shouldSaveProject;
-
-					if (this.$staticConfig.triggerJsonSchemaValidation) {
-						this.$jsonSchemaValidator.validate(this.projectData);
-					}
-
-					let debugProjectFile = path.join(projectDir, this.$projectConstants.DEBUG_PROJECT_FILE_NAME);
-					if (this.$options.debug && !this.$fs.exists(debugProjectFile).wait()) {
-						this.$fs.writeJson(debugProjectFile, {}).wait();
-					}
-
-					let releaseProjectFile = path.join(projectDir, this.$projectConstants.RELEASE_PROJECT_FILE_NAME);
-					if (this.$options.release && !this.$fs.exists(releaseProjectFile).wait()) {
-						this.$fs.writeJson(releaseProjectFile, {}).wait();
-					}
-
-					let allProjectFiles = this.$fs.enumerateFilesInDirectorySync(projectDir, (file: string, stat: IFsStats) => {
-						return Project.CONFIGURATION_FILE_SEARCH_PATTERN.test(file);
-					});
-
-					_.each(allProjectFiles, (configProjectFile: string) => {
-						let configMatch = path.basename(configProjectFile).match(Project.CONFIGURATION_FROM_FILE_NAME_REGEX);
-						if (configMatch && configMatch.length > 1) {
-							let configurationName = configMatch[1];
-							let configProjectContent = this.$fs.readJson(configProjectFile).wait();
-							this.configurationSpecificData[configurationName.toLowerCase()] = configProjectContent;
-							this._hasBuildConfigurations = true;
-						}
-					});
-				} catch (err) {
-					if (err === "FUTURE_PROJECT_VER") {
-						this.$errors.fail({
-							formatStr: "This project is created by a newer version of AppBuilder. Upgrade AppBuilder CLI to work with it.",
-							suppressCommandHelp: true
-						});
+		return fullPath.substring(projectDir.length);
```

Comment:
don't think so, but even if that were the case, then an error would be thrown a little above, because
`_.startsWith(fullPath, projectDir)` would return `false`